crates/sui-json-rpc/src/error.rs

```diff
@@ -156,3 +241,207 @@ impl From<SuiRpcInputError> for RpcError {
+                retried_tx_success: Some(true),
+            };
+
+            let rpc_error = match_quorum_driver_error(quorum_driver_error);
+
+            let error_object: ErrorObjectOwned = rpc_error.into();
+            let expected_code = expect!["-32602"];
+            expected_code.assert_eq(&error_object.code().to_string());
+            let expected_message = expect!["Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction Some(TransactionDigest(11111111111111111111111111111111)), success: Some(true)"];
+            expected_message.assert_eq(error_object.message());
+            let expected_data = expect![[
+                r#"{"11111111111111111111111111111111":[["0xaa7c752529db203b5bf37795d37bb998650e8baba9302441fcb1eebb682942a8",0,"11111111111111111111111111111111"]]}"#
+            ]];
+            let actual_data = error_object.data().unwrap().to_string();
+            expected_data.assert_eq(&actual_data);
+        }
+
+        #[test]
+        fn test_non_recoverable_transaction_error() {
+            let quorum_driver_error = QuorumDriverError::NonRecoverableTransactionError {
+                errors: vec![
+                    (
+                        SuiError::UserInputError {
+                            error: UserInputError::GasBalanceTooLow {
+                                gas_balance: 10,
+                                needed_gas_amount: 100,
+                            },
+                        },
+                        0,
+                        vec![],
+                    ),
+                    (
+                        SuiError::UserInputError {
+                            error: UserInputError::ObjectVersionUnavailableForConsumption {
+                                provided_obj_ref: random_object_ref(),
+                                current_version: 10.into(),
+                            },
+                        },
+                        0,
+                        vec![],
+                    ),
+                ],
+            };
+
+            let rpc_error = match_quorum_driver_error(quorum_driver_error);
+
+            let error_object: ErrorObjectOwned = rpc_error.into();
+            let expected_code = expect!["-32001"];
+            expected_code.assert_eq(&error_object.code().to_string());
+            let expected_message =
+                expect!["Transaction execution failed due to issues with transaction inputs. Please review the 'data' field and try again."];
+            expected_message.assert_eq(error_object.message());
+            let expected_data = expect![[
+                r#"["Balance of gas object 10 is lower than the needed amount: 100.","Object (0x88d272a6773e6b8ed57f28b9e0eabb45f34ba39aaa1e88610ef242f0847071a6, SequenceNumber(0), o#11111111111111111111111111111111) is not available for consumption, its current version: SequenceNumber(10)."]"#
```

Comment:
Can be a follow-up, but the "not available for consumption" error message could be made a little clearer by reducing some of the information and cutting down the information:

- Do we need to mention its digest?
- Can we represent versions in a format that's more familiar to people?  `SequenceNumber` is the term we use for them internally, but most people know them as versions).

Reply:
* As in we should mention its digest or we should drop it?
* Yeah that makes sense to me

Reply:
> As in we should mention its digest or we should drop it?

We currently display the digest, because we are displaying the whole object reference in debug format.  The question is whether we can drop it

Reply:
I see, yeah it doesn't look like we use object digest generally on the rpc, so we can probably afford to drop that information.

Reply:
Will address in a separate PR as this would be a change to SuiError